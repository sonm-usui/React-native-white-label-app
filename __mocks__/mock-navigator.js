// @flow
import * as React from 'react'
import { NavigationContainer } from '@react-navigation/native'
import { createStackNavigator } from '@react-navigation/stack'

jest.mock('react-native-screens', () => {
  const RealComponent = jest.requireActual('react-native-screens')
  RealComponent.enableScreens = jest.fn()
  return RealComponent
})

const Stack = createStackNavigator()
export const MockedNavigator = ({
  component,
  params = {},
}: {
  component: React.ElementType,
  params?: Object,
}) => {
  return (
    <NavigationContainer>
      <Stack.Navigator>
        <Stack.Screen name="MockedScreen" initialParams={params}>
          {component}
        </Stack.Screen>
      </Stack.Navigator>
    </NavigationContainer>
  )
}
