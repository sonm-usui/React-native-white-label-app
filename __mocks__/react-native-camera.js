// @flow
import React, { PureComponent } from 'react'

export class RNCamera extends PureComponent<any, any> {
  static Constants = {
    Aspect: {},
    BarCodeType: {},
    Type: { back: 'back', front: 'front' },
    CaptureMode: {},
    CaptureTarget: {},
    CaptureQuality: {},
    Orientation: {},
    FlashMode: {},
    TorchMode: {},
  }

  render() {
    return this.props.children
  }
}

export default RNCamera
