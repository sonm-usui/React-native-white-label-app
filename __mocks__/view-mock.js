// @flow
import React, { PureComponent } from 'react'
import { View } from 'react-native'

// We don't need to type check external library
// and we don't need types for mock test stub
class MockView extends PureComponent<any, void> {
  render() {
    return <View {...this.props}>{this.props.children}</View>
  }
}

export default MockView
