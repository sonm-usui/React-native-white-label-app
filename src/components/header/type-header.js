// @flow

import { verticalScale, moderateScale } from 'react-native-size-matters'
import { StyleSheet, Platform } from 'react-native'
import { colors, fontFamily } from '../../common/styles/constant'

export type HeaderProps = {
  headline?: string,
  hideBackButton?: boolean,
  transparent?: boolean,
  showUnreadMessagesBadge?: string,
  color?: string,
  additionalActionOnBackPress?: () => void,
}

export const SHOW_UNREAD_MESSAGES_BADGE_NEAR_WITH_MENU = 'menu'
export const SHOW_UNREAD_MESSAGES_BADGE_NEAR_WITH_TITLE = 'title'

export const styles = StyleSheet.create({
  container: {
    zIndex: 1,
    height: Platform.OS === 'android' ? verticalScale(56) : verticalScale(80),
    width: '100%',
    flexDirection: 'row',
    backgroundColor: colors.white,
    shadowColor: colors.black,
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.2,
    shadowRadius: 10,
    elevation: Platform.OS === 'android' ? 8 : 0,
    paddingTop: Platform.OS === 'android' ? 0 : verticalScale(24),
  },
  containerTransparent: {
    height: Platform.OS === 'android' ? verticalScale(56) : verticalScale(80),
    width: '100%',
    flexDirection: 'row',
    paddingTop: Platform.OS === 'android' ? 0 : verticalScale(24),
  },
  headerTransparent: {
    position: 'absolute',
    backgroundColor: 'transparent',
    zIndex: 100,
    top: 0,
    left: 0,
    right: 0,
    elevation: 0,
    shadowOpacity: 0,
    borderBottomWidth: 0,
  },
  iconSection: {
    flexDirection: 'row',
    alignItems: 'center',
    width: '15%',
  },
  labelSection: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    width: '85%',
  },
  menuIcon: {
    marginLeft: moderateScale(15),
    marginBottom: moderateScale(2),
  },
  label: {
    fontFamily: fontFamily,
    fontSize: moderateScale(22),
    marginLeft: moderateScale(-50),
    color: colors.gray2,
  },
  labelForOptions: {
    fontFamily: fontFamily,
    fontSize: moderateScale(22),
    marginRight: moderateScale(32 + 15),
    textAlign: 'center',
    color: colors.gray2,
  },
  iconAndNameWrapper: {
    width: '65%',
    height: '100%',
    marginStart: moderateScale(-8),
    marginEnd: moderateScale(32),
    flexDirection: 'row',
  },
  labelWithIconSection: {
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'center',
    width: '85%',
  },
  headerImageWrapper: {
    width: moderateScale(32),
    height: moderateScale(32),
    borderRadius: 20,
  },
  headerIcon: {
    width: moderateScale(32),
    height: moderateScale(32),
  },
  buttonMoreOptionsWrapper: {
    height: '100%',
    width: '15%',
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'row',
  },
  headerImageOuterWrapper: {
    height: '100%',
    width: '35%',
    alignItems: 'center',
    flexDirection: 'row',
    marginLeft: moderateScale(10),
    marginRight: moderateScale(6),
    paddingBottom: moderateScale(6),
  },
})
