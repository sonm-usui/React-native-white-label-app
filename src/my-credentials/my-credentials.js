// @flow
import React, { useMemo, useCallback, useEffect } from 'react'
import { View, StyleSheet } from 'react-native'
import { bindActionCreators } from 'redux'
import { connect, useSelector, useDispatch } from 'react-redux'
import type { MyCredentialsProps, CredentialItem } from './type-my-credentials'

import { CameraButton } from '../components'
import { CredentialsCards } from './cards/credentials-cards'
import { myCredentialsRoute, qrCodeScannerTabRoute } from '../common'
import { colors } from '../common/styles/constant'
import { getReceivedCredentials } from '../store/store-selector'
import { EmptyState } from '../home/empty-state'
import {
  credentialsHeadline,
  credentialsShowCameraButton,
  CustomMyCredentialsScreen,
  MyCredentialsViewEmptyState,
} from '../external-imports'
import { deleteClaim } from '../claim-offer/claim-offer-store'

export const headlineForCredentialRoute =
  credentialsHeadline || 'MY Credentials'
const showCameraButton =
  typeof credentialsShowCameraButton === 'boolean'
    ? credentialsShowCameraButton
    : true

const MyCredentialsScreen = ({ route, navigation }: MyCredentialsProps) => {
  const receivedCredentials = useSelector(getReceivedCredentials)
  const dispatch = useDispatch()
  const deleteCredential = useCallback(
    (uuid: string) => dispatch(deleteClaim(uuid)),
    [dispatch]
  )

  const credentials = useMemo(() => {
    const credentials: Array<CredentialItem> = receivedCredentials.map(
      (credential) => ({
        claimOfferUuid: credential.uid,
        credentialName: credential.data.name,
        issuerName: credential.issuer.name,
        date: credential.issueDate,
        attributes: credential.data.revealedAttributes,
        logoUrl: credential.senderLogoUrl,
        remoteDid: credential.remotePairwiseDID,
        colorTheme: credential.colorTheme,
        claimDefinitionId: credential.data.claimDefinitionId,
      })
    )

    credentials.sort((a, b) =>
      a.credentialName
        .toLowerCase()
        .localeCompare(b.credentialName.toLowerCase())
    )

    return credentials
  }, [receivedCredentials])

  const hasNoCredentials = credentials.length === 0

  return (
    <View style={styles.outerContainer}>
      <View style={styles.container}>
        {hasNoCredentials &&
          (MyCredentialsViewEmptyState ? (
            <MyCredentialsViewEmptyState />
          ) : (
            <EmptyState />
          ))}
        {!hasNoCredentials && (
          <CredentialsCards
            credentials={credentials}
            deleteClaim={deleteCredential}
            navigation={navigation}
            route={route}
          />
        )}
      </View>

      {showCameraButton && (
        <CameraButton
          onPress={() => navigation.navigate(qrCodeScannerTabRoute)}
        />
      )}
    </View>
  )
}

export const MyCredentials = CustomMyCredentialsScreen || MyCredentialsScreen

export const myCredentialsScreen = {
  routeName: myCredentialsRoute,
  screen: MyCredentials,
}

const styles = StyleSheet.create({
  outerContainer: {
    flex: 1,
  },
  container: {
    backgroundColor: colors.white,
    flex: 1,
  },
  backgroundImage: {
    width: '100%',
    height: '100%',
    alignItems: 'center',
    justifyContent: 'center',
  },
})
