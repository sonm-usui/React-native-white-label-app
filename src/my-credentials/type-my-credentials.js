// @flow
import type { ReactNavigation } from '../common/type-common'
import type { ClaimOfferPayload } from '../claim-offer/type-claim-offer'
import type { Attribute } from '../push-notification/type-push-notification'
import { deleteClaim } from '../claim-offer/claim-offer-store'

export type MyCredentialsProps = {
  offers: ClaimOffers,
  environmentName: string,
  deleteClaim: typeof deleteClaim,
  credentialsColors: any,
} & ReactNavigation

export type CredentialsCardsProps = {
  credentials: Array<CredentialItem>,
} & ReactNavigation

export type CredentialCardProps = {
  item: CredentialItem,
  isExpanded: boolean,
  isHidden: boolean,
  setActiveStack?: (stackName: string | null) => void,
  elevation?: number,
  enabled?: boolean,
  isNeedMargin?: boolean,
}

export type CredentialItem = {
  claimOfferUuid: string,
  credentialName: string,
  issuerName: string,
  date?: number,
  attributes: Array<Attribute>,
  logoUrl?: ?string,
  remoteDid: string,
  colorTheme?: string,
  claimDefinitionId?: ?string,
}

export type ClaimOffers = {
  +[string]: ClaimOfferPayload,
}

export type CardStackProps = {
  credentials: Array<CredentialItem>,
  isExpanded: boolean,
  isHidden: boolean,
  setActiveStack: (stackName: string | null, animate?: boolean) => void,
  isNeedMargin?: boolean,
}

export const MESSAGE_DELETE_CLAIM_TITLE = 'Delete credential?'
export const MESSAGE_DELETE_CLAIM_DESCRIPTION = 'This cannot be undone.'
