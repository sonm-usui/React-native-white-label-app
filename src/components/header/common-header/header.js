// @flow
import React from 'react'
import { View, TouchableOpacity, Text, Platform } from 'react-native'

import type { HeaderProps } from '../type-header'
import { styles } from '../type-header'

import {
  EvaIcon,
  ANDROID_BACK_ARROW_ICON,
  IOS_BACK_ARROW_ICON,
} from '../../../common/icons'
import { moderateScale } from 'react-native-size-matters'

export const Header = (props: HeaderProps) => {
  const iOS = Platform.OS === 'ios'
  const {
    transparent,
    hideBackButton,
    headline,
    additionalActionOnBackPress,
  } = props

  const onPress = () => {
    goBack(props)
    additionalActionOnBackPress && additionalActionOnBackPress()
  }

  return (
    <View style={transparent ? styles.containerTransparent : styles.container}>
      <View
        accessible={true}
        accessibilityLabel="left-icon"
        style={styles.iconSection}
      >
        {!hideBackButton && (
          <TouchableOpacity testID="left-icon" onPress={onPress}>
            <EvaIcon
              name={iOS ? IOS_BACK_ARROW_ICON : ANDROID_BACK_ARROW_ICON}
              width={moderateScale(32)}
              height={moderateScale(32)}
              style={styles.menuIcon}
              {...(props.color ? { color: props.color } : {})}
            />
          </TouchableOpacity>
        )}
      </View>
      <View style={styles.labelSection}>
        <Text style={styles.label}>{headline}</Text>
      </View>
    </View>
  )
}

const goBack = (props: any) => {
  const { navigation } = props
  const backRedirectRoute = props.route && props.route.params?.backRedirectRoute
  if (backRedirectRoute) {
    navigation.navigate(backRedirectRoute)
  } else {
    navigation.goBack(null)
  }
}
