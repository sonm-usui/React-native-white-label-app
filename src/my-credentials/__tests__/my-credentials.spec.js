// @flow

import 'react-native'
import React from 'react'
import { Platform } from 'react-native'
import { getStore, getNavigation } from '../../../__mocks__/static-data'
import renderer, { act } from 'react-test-renderer'
import { MyCredentials } from '../my-credentials'
import { Provider } from 'react-redux'
import { MockedNavigator } from '../../../__mocks__/mock-navigator'
import { SHOW_UNREAD_MESSAGES_BADGE_NEAR_WITH_MENU } from '../../components/header/type-header'
import { defaultEnvironment } from '../../environment'

describe('my credentials screen', () => {
  jest.useFakeTimers()

  const store = getStore()

  function getProps() {
    return {
      navigation: getNavigation(),
      claimMap: {
        uuid_1: {
          claimOfferUuid: 'offer_uuid_1',
          senderDID: 'senderDID_1',
          myPairwiseDID: 'myPairwiseDID_1',
          logoUrl: 'https://logourl.com/logo.png',
          issueDate: 1,
        },
      },
      offers: {
        offer_uuid_1: {
          remotePairwiseDID: 'senderDID_1',
          status: 'RECEIVED',
          claimRequestStatus: 'CLAIM_REQUEST_SUCCESS',
          data: {
            name: 'credential_name',
            version: '3.0',
            revealedAttributes: [
              {
                label: 'attribute_1',
              },
              {
                label: 'attribute_2',
              },
              {
                label: 'attribute_3',
              },
            ],
            claimDefinitionSchemaSequenceNumber: 1,
          },
          issuer: {
            name: 'issuer_name',
            did: 'uuid_1',
          },
        },
      },
      environmentName: defaultEnvironment,
      route: {},
      showUnreadMessagesBadge: { SHOW_UNREAD_MESSAGES_BADGE_NEAR_WITH_MENU },
    }
  }

  function setup() {
    const props = getProps()
    return { props }
  }

  function getMockedNavigator(props) {
    const component = () => (
      <Provider store={store}>
        <MyCredentials {...props} />
      </Provider>
    )

    return <MockedNavigator component={component} />
  }

  function render(props) {
    const wrapper = renderer.create(getMockedNavigator(props))
    act(() => jest.runAllTimers())
    const componentInstance = wrapper.root.findByType(MyCredentials).instance
    return { wrapper, componentInstance }
  }

  xit('should render properly and snapshot should match ios platform', () => {
    // TODO:KS Need to fix date issue on CI server. Meanwhile commenting this test
    const { props } = setup()
    const { wrapper } = render(props)
    const tree = wrapper.toJSON()
    expect(tree).toMatchSnapshot()
  })

  xit('should render properly and snapshot should match for android platform', () => {
    // skipping this test, somehow react-navigation v5 is not working correctly with jest on Android
    const existingOS = Platform.OS
    Platform.OS = 'android'
    const { props } = setup()
    const { wrapper } = render(props)
    const tree = wrapper.toJSON()
    expect(tree).toMatchSnapshot()
    // revert environment to what it was before
    Platform.OS = existingOS
  })
})
