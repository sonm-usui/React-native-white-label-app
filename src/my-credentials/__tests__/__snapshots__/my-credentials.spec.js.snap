// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my credentials screen should render properly and snapshot should match ios platform 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "zIndex": 1,
          },
          false,
        ]
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgb(255, 255, 255)",
                "borderBottomColor": "rgb(224, 224, 224)",
                "flex": 1,
                "shadowColor": "rgb(224, 224, 224)",
                "shadowOffset": Object {
                  "height": 0.5,
                  "width": 0,
                },
                "shadowOpacity": 0.85,
                "shadowRadius": 0,
              }
            }
          />
        </View>
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 64,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 20,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "marginHorizontal": 16,
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 17,
                    "fontWeight": "600",
                  }
                }
              >
                MockedScreen
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
                "opacity": 1,
              }
            }
          >
            <View
              enabled={false}
              hitSlop={
                Object {
                  "right": -270,
                }
              }
              maxDeltaY={20}
              minOffsetX={5}
              onHandlerStateChange={[Function]}
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "rgb(242, 242, 242)",
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": -1,
                      },
                      "shadowOpacity": 0.3,
                      "shadowRadius": 5,
                      "top": 0,
                      "width": 3,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "flex": 1,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "backgroundColor": "#FFFFFF",
                              "height": "100%",
                              "width": "100%",
                            }
                          }
                          testID="my-credentials-container"
                          accessibilityLabel="my-credentials-container"
                          accessible="true"
                        >
                          <RCTScrollView
                            contentContainerStyle={
                              Object {
                                "paddingTop": 71.47058823529412,
                              }
                            }
                            data={
                              Array [
                                Object {
                                  "attributesCount": 3,
                                  "claimUuid": "senderDID_1",
                                  "credentialName": "credential_name",
                                  "date": 1,
                                  "logoUrl": "https://logourl.com/logo.png",
                                },
                              ]
                            }
                            disableVirtualization={false}
                            getItem={[Function]}
                            getItemCount={[Function]}
                            horizontal={false}
                            initialNumToRender={10}
                            keyExtractor={[Function]}
                            maxToRenderPerBatch={10}
                            numColumns={1}
                            onContentSizeChange={[Function]}
                            onEndReachedThreshold={2}
                            onLayout={[Function]}
                            onMomentumScrollEnd={[Function]}
                            onScroll={[Function]}
                            onScrollBeginDrag={[Function]}
                            onScrollEndDrag={[Function]}
                            removeClippedSubviews={false}
                            renderItem={[Function]}
                            scrollEventThrottle={50}
                            stickyHeaderIndices={Array []}
                            style={
                              Object {
                                "backgroundColor": "#FFFFFF",
                                "height": "100%",
                                "width": "100%",
                              }
                            }
                            updateCellsBatchingPeriod={50}
                            viewabilityConfigCallbackPairs={Array []}
                            windowSize={21}
                          >
                            <View>
                              <View
                                onLayout={[Function]}
                                style={null}
                              >
                                <View
                                  accessible={true}
                                  focusable={false}
                                  isTVSelectable={true}
                                  onClick={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                  style={
                                    Object {
                                      "borderBottomColor": "#F2F2F2",
                                      "borderBottomWidth": 1,
                                      "flexDirection": "row",
                                      "height": 86.11428571428571,
                                      "opacity": 1,
                                      "width": "100%",
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "height": "100%",
                                        "paddingTop": 19.058823529411764,
                                        "width": 64,
                                      }
                                    }
                                  >
                                    <View
                                      accessibilityLabel="credential_name-avatar"
                                      accessible={true}
                                      imageShadow={null}
                                      style={
                                        Object {
                                          "borderRadius": 50,
                                        }
                                      }
                                      testID="credential_name-avatar"
                                    >
                                      <Image
                                        accessibilityLabel="credential_name-avatar-image"
                                        accessible={true}
                                        focusable={true}
                                        onClick={[Function]}
                                        onResponderGrant={[Function]}
                                        onResponderMove={[Function]}
                                        onResponderRelease={[Function]}
                                        onResponderTerminate={[Function]}
                                        onResponderTerminationRequest={[Function]}
                                        onStartShouldSetResponder={[Function]}
                                        source={
                                          Object {
                                            "uri": "https://logourl.com/logo.png",
                                          }
                                        }
                                        style={
                                          Array [
                                            Object {
                                              "height": 32,
                                              "width": 32,
                                            },
                                            Object {
                                              "borderRadius": 16,
                                            },
                                          ]
                                        }
                                        testID="credential_name-avatar-image"
                                      />
                                    </View>
                                  </View>
                                  <View
                                    style={
                                      Object {
                                        "flex": 1,
                                      }
                                    }
                                  >
                                    <View
                                      style={
                                        Object {
                                          "flex": 1,
                                          "flexDirection": "row",
                                          "height": "50%",
                                        }
                                      }
                                    >
                                      <View
                                        style={
                                          Object {
                                            "height": "100%",
                                            "justifyContent": "flex-end",
                                          }
                                        }
                                      >
                                        <Text
                                          ellipsizeMode="tail"
                                          numberOfLines={1}
                                          style={
                                            Object {
                                              "color": "#505050",
                                              "fontFamily": "Lato",
                                              "fontSize": 11.911764705882351,
                                              "fontWeight": "bold",
                                            }
                                          }
                                        >
                                          credential_name
                                        </Text>
                                      </View>
                                    </View>
                                    <View
                                      style={
                                        Object {
                                          "flex": 1,
                                          "height": "50%",
                                        }
                                      }
                                    >
                                      <View
                                        style={
                                          Object {
                                            "height": "100%",
                                            "width": "96%",
                                          }
                                        }
                                      >
                                        <Text
                                          ellipsizeMode="tail"
                                          numberOfLines={1}
                                          style={
                                            Object {
                                              "color": "#777777",
                                              "fontFamily": "Lato",
                                              "fontSize": 8.735294117647058,
                                            }
                                          }
                                        >
                                          3 attributes
                                        </Text>
                                      </View>
                                    </View>
                                  </View>
                                  <View
                                    style={
                                      Object {
                                        "alignItems": "flex-end",
                                        "flexDirection": "row",
                                        "height": "100%",
                                        "width": "32%",
                                      }
                                    }
                                  >
                                    <View
                                      style={
                                        Object {
                                          "alignItems": "center",
                                          "flexDirection": "row",
                                          "height": "100%",
                                          "justifyContent": "flex-end",
                                          "width": "85%",
                                        }
                                      }
                                    >
                                      <Text
                                        style={
                                          Object {
                                            "color": "#A5A5A5",
                                            "fontFamily": "Lato",
                                            "fontSize": 7.941176470588235,
                                            "fontStyle": "italic",
                                          }
                                        }
                                      >
                                        02/13/2009
                                      </Text>
                                    </View>
                                  </View>
                                </View>
                              </View>
                            </View>
                          </RCTScrollView>
                        </View>
                        <View
                          style={
                            Object {
                              "backgroundColor": "#FFFFFF",
                              "elevation": 0,
                              "flexDirection": "row",
                              "height": 71.47058823529412,
                              "position": "absolute",
                              "shadowColor": "#000000",
                              "shadowOffset": Object {
                                "height": 2,
                                "width": 0,
                              },
                              "shadowOpacity": 0.2,
                              "shadowRadius": 10,
                              "top": 0,
                              "width": "100%",
                              "zIndex": 1,
                            }
                          }
                        >
                          <View
                            accessibilityLabel="burger-menu"
                            accessible={true}
                            style={
                              Object {
                                "alignItems": "flex-end",
                                "flexDirection": "row",
                                "paddingBottom": 9.571428571428571,
                                "width": "15%",
                              }
                            }
                          >
                            <View
                              accessible={true}
                              focusable={true}
                              isTVSelectable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Object {
                                  "opacity": 1,
                                }
                              }
                              testID="burger-menu"
                            >
                              <RNSVGSvgView
                                align="xMidYMid"
                                bbHeight="30.628571428571426"
                                bbWidth="30.628571428571426"
                                focusable={false}
                                height="30.628571428571426"
                                meetOrSlice={0}
                                minX={0}
                                minY={0}
                                style={
                                  Array [
                                    Object {
                                      "backgroundColor": "transparent",
                                      "borderWidth": 0,
                                    },
                                    Object {
                                      "marginLeft": 14.357142857142858,
                                    },
                                    Object {
                                      "flex": 0,
                                      "height": 30,
                                      "width": 30,
                                    },
                                  ]
                                }
                                vbHeight={32}
                                vbWidth={32}
                                width="30.628571428571426"
                              >
                                <RNSVGGroup>
                                  <RNSVGPath
                                    d="M5.264 8H26.7347C27.4307 8 28 8.56933 28 9.26533V9.40133C28 10.0973 27.4307 10.6667 26.7347 10.6667H5.264C4.568 10.6667 4 10.0973 4 9.40133V9.26533C4 8.56933 4.568 8 5.264 8ZM26.7347 14.6667H5.264C4.568 14.6667 4 15.236 4 15.932V16.068C4 16.764 4.568 17.3333 5.264 17.3333H26.7347C27.4307 17.3333 28 16.764 28 16.068V15.932C28 15.236 27.4307 14.6667 26.7347 14.6667ZM26.7347 21.3333H5.264C4.568 21.3333 4 21.9027 4 22.5987V22.7347C4 23.4307 4.568 24 5.264 24H26.7347C27.4307 24 28 23.4307 28 22.7347V22.5987C28 21.9027 27.4307 21.3333 26.7347 21.3333Z"
                                    fill={4286019447}
                                    fillRule={0}
                                    propList={
                                      Array [
                                        "fill",
                                        "fillRule",
                                      ]
                                    }
                                  />
                                </RNSVGGroup>
                              </RNSVGSvgView>
                            </View>
                          </View>
                          <View
                            style={
                              Object {
                                "alignItems": "flex-end",
                                "flexDirection": "row",
                                "justifyContent": "center",
                                "paddingBottom": 14.357142857142858,
                                "width": "85%",
                              }
                            }
                          >
                            <Text
                              style={
                                Object {
                                  "color": "#777777",
                                  "fontFamily": "Lato",
                                  "fontSize": 21.057142857142857,
                                  "marginLeft": -47.85714285714286,
                                }
                              }
                            >
                              My Credentials
                            </Text>
                          </View>
                        </View>
                        <View
                          accessible={true}
                          focusable={true}
                          isTVSelectable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#FFFFFF",
                              "borderRadius": 30,
                              "bottom": 20,
                              "elevation": 0,
                              "height": 31.199999999999996,
                              "justifyContent": "center",
                              "opacity": 1,
                              "position": "absolute",
                              "right": 20,
                              "shadowColor": "#000000",
                              "shadowOffset": Object {
                                "height": 2,
                                "width": 0,
                              },
                              "shadowOpacity": 0.5,
                              "shadowRadius": 5,
                              "width": 31.199999999999996,
                              "zIndex": 1000,
                            }
                          }
                        >
                          <RNSVGSvgView
                            align="xMidYMid"
                            bbHeight="21.057142857142857"
                            bbWidth="21.057142857142857"
                            focusable={false}
                            height="21.057142857142857"
                            meetOrSlice={0}
                            minX={0}
                            minY={0}
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "transparent",
                                  "borderWidth": 0,
                                },
                                Object {
                                  "flex": 0,
                                  "height": 21,
                                  "width": 21,
                                },
                              ]
                            }
                            vbHeight={24}
                            vbWidth={24}
                            width="21.057142857142857"
                          >
                            <RNSVGGroup>
                              <RNSVGPath
                                d="M16.83 5H20C21.1 5 22 5.9 22 7V19C22 20.1 21.1 21 20 21H4C2.9 21 2 20.1 2 19V7C2 5.9 2.9 5 4 5H7.17L9 3H15L16.83 5ZM4 19H20V7H15.95L14.12 5H9.88L8.05 7H4V19ZM12 8C9.24 8 7 10.24 7 13C7 15.76 9.24 18 12 18C14.76 18 17 15.76 17 13C17 10.24 14.76 8 12 8ZM9 13C9 14.65 10.35 16 12 16C13.65 16 15 14.65 15 13C15 11.35 13.65 10 12 10C10.35 10 9 11.35 9 13Z"
                                fill={4286019447}
                                fillRule={0}
                                propList={
                                  Array [
                                    "fill",
                                    "fillRule",
                                  ]
                                }
                              />
                            </RNSVGGroup>
                          </RNSVGSvgView>
                          <Text
                            style={
                              Object {
                                "color": "#777777",
                                "fontFamily": "Lato",
                                "fontSize": 8.922857142857143,
                              }
                            }
                          >
                            Scan
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
